data-reconciliation-app/src/services/authentication-service.ts

```diff
@@ -127,34 +208,75 @@ export class CertBasedAuthenticationService implements IAuthenticationService {
     }
   }
 
-  // 
-  public isValidIdentity(identityId: string): ServiceResult<boolean> {
-    if (!identityId) {
+  /**
+   * Check if caller's access token is valid
+   * @param {JwtAuthnIdentity} identity JwtAuthnIdentity object
+   * @returns {ServiceResult<boolean>}
+   */
+  public isValidJwtToken(identity: ccfapp.JwtAuthnIdentity): ServiceResult<boolean> {
+
```

Comment:
Sadly CCF doc says the app needs to validate the signature of JWT token. I think we need to add the code for it.

> CCF provides support for managing public token signing keys required for validating tokens. In the future, CCF will support validating token signatures natively (currently the responsibility of apps).

https://microsoft.github.io/CCF/main/build_apps/auth/jwt.html

Reply:
Do we have a sample or documentation that describes how to retrieve the JWT issuer signing keys from CCF to use them for validating the token signature?

Reply:
I've created a new issue to capture it #193

Reply:
There is a description in the link I put above.

> Validating a token means checking its format, signature, and IdP- and app-specific claims. See [tests/js-authentication/src/endpoints.js](https://github.com/microsoft/CCF/tree/main/tests/js-authentication/src/endpoints.js) for an example on how to do this in TypeScript.

But it's weird to me because I don't find any code verifying the signature.

Based on the documentation, I think there should be code like [this](https://github.com/microsoft/CCF/blob/ccf-3.0.5/tests/npm-app/src/endpoints/jwt.ts#L71).

@achamayou Could you tell us what we should here? (Do apps need to write code like `if (verifyJWT(publicKey, jwt) {...}` by themselves?)

Reply:
@takuro-sato, Based on my conversation with @jumaffre, CCF now supports token signature verification by default. but the app needs to check the payload of the JWT in the app to handle authorisation

Reply:
This is the native code that handles this: https://github.com/microsoft/CCF/blob/main/src/endpoints/authentication/jwt_auth.cpp#L13 I've updated the docs in https://github.com/microsoft/CCF/pull/4936